### PR TITLE
Disable asm-processor by default

### PR DESCRIPTION
**Pull request by encounter**
With `--non-matching`, asm-processor is enabled to allow modding builds. CI runs both matching and non-matching builds.

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -136,6 +136,7 @@
 config.generate_map = args.map
 config.sjiswrap_path = args.sjiswrap
 config.non_matching = args.non_matching
+config.asm_processor = args.non_matching
```
Comment (Yanis42) [inline configure.py:139]: I think it's odd to use ``args.non_matching`` to set the asm_processor variable

thinking about it more, it makes sense but idk it looks weird but feel free to ignore

File: tools/project.py [hunk 2]
```diff
@@ -888,7 +897,11 @@ def add_unit(build_obj, link_step: LinkStep):
                 link_step.add(obj_path)
                 return
 
-            link_built_obj = obj.completed(version)
+            link_built_obj = obj.completed(version) and (
+                # Don't link asm_processor objects if asm_processor is disabled
+                config.asm_processor
+                or not obj.options["asm_processor"]
+            )
```
Comment (cadmic) [inline tools/project.py:904]: this broke my brain. How about
```python
link_built_obj = obj.completed(version)
# Don't link asm_processor objects if asm_processor is disabled
if obj.options["asm_processor"] and not config.asm_processor:
    link_built_obj = False
```

File: tools/project.py [hunk 3]
```diff
@@ -1227,12 +1243,14 @@ def keep_flag(flag):
...
             progress_categories.append(f"{version}.{category_opt}")
-        unit_config["metadata"].update({
-            "complete": obj.completed(version),
-            "reverse_fn_order": reverse_fn_order,
-            "source_path": obj.src_path,
-            "progress_categories": progress_categories,
-        })
+        unit_config["metadata"].update(
+            {
+                "complete": obj.completed(version) and not obj.options["asm_processor"],
```
Comment (cadmic) [inline tools/project.py:1248]: what exactly does `"complete"` mean? should this use the same logic as `link_built_obj`?

**Review by Yanis42 — COMMENTED** [review]
building with ``--non-matching`` seems to fail (maybe windows only?)

<details>

```
PS F:\github\oot-gc> python configure.py --non-matching
PS F:\github\oot-gc> ninja
[1/154] MWCC build\ce-j\src\emulator\mcardGCN.o
FAILED: build/ce-j/src/emulator/mcardGCN.o 
set "BINUTILS_PATH=build\binutils" && sh tools/asm_processor/compile.sh "build\tools\sjiswrap.exe build\compilers\GC\1.1\mwcceppc.exe -Cpp_exceptions off -proc gekko -fp hardware -fp_contract on -enum int -align powerpc -nosyspath -RTTI off -str reuse -multibyte -O4,p -inline auto -nodefaults -msgstyle gcc -sym on -i include -i libc -DNON_MATCHING -DASM_PROCESSOR -inline deferred -i build/ce-j/include -DVERSION=CE_J -MMD" "build\binutils\powerpc-eabi-as.exe -mgekko -I include -I libc" src\emulator\mcardGCN.c build\ce-j\src\emulator\mcardGCN.o
CreateProcess failed: The system cannot find the file specified.
[2/154] MWCC build\ce-j\src\emulator\frame.o
FAILED: build/ce-j/src/emulator/frame.o 
set "BINUTILS_PATH=build\binutils" && sh tools/asm_processor/compile.sh "build\tools\sjiswrap.exe build\compilers\GC\1.1\mwcceppc.exe -Cpp_exceptions off -proc gekko -fp hardware -fp_contract on -enum int -align powerpc -nosyspath -RTTI off -str reuse -multibyte -O4,p -inline auto -nodefaults -msgstyle gcc -sym on -i include -i libc -DNON_MATCHING -DASM_PROCESSOR -inline deferred -i build/ce-j/include -DVERSION=CE_J -MMD" "build\binutils\powerpc-eabi-as.exe -mgekko -I include -I libc" src\emulator\frame.c build\ce-j\src\emulator\frame.o
CreateProcess failed: The system cannot find the file specified.
[14/154] MWCC build\ce-j\src\emulator\THPVideoDecode.oninja: fatal: ReadFile: GetLastError() = 6
```

</details>

**Comment by cadmic** [comment]
superseded by #185

**Pull request closed without merging**